pkg/kubelet/kubelet.go

```diff
@@ -1500,7 +1498,35 @@ func (kl *Kubelet) initializeRuntimeDependentModules() {
 func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	ctx := context.Background()
 	if kl.logServer == nil {
-		kl.logServer = http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/")))
+		file := http.FileServer(http.Dir("/var/log/"))
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeLogQuery) && kl.kubeletConfiguration.EnableSystemLogQuery {
+			kl.logServer = http.StripPrefix("/logs/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+				query := req.URL.Query()
+				nlq, errs := newNodeLogQuery(query)
```

Comment:
it would be clearer to make the query handling code path completely separate and mutually exclusive from file handling, so that we're _either_ operating in query mode or in path mode

```go
if nlq, errs := newNodeLogQuery(query); len(errs) == 0 {
  ... handle query parse error
  return
} else if nlq != nil {
  if req.URL.Path != "/" {
    ... error, no path allowed in query mode
    return
  }
  if errs := nlq.validate(); len(errs) > 0 {
    ...
    return
  }
  ... query mode serving
  return
}

// fall back to file mode handling
file.ServeHTTP(w, req)
```